application/forms/ChannelForm.php

```diff
@@ -152,50 +153,77 @@ public function hasBeenSubmitted()
         return parent::hasBeenSubmitted();
     }
 
-    public function populate($values)
+    /**
+     * Load the channel with given id
+     *
+     * @param int $id
+     *
+     * @return $this
+     *
+     * @throws HttpNotFoundException
+     */
+    public function loadChannel(int $id): self
     {
-        if ($values instanceof Channel) {
-            $values = [
-                'name'      => $values->name,
-                'type'      => $values->type,
-                'config'    => json_decode($values->config, true) ?? []
-            ];
-        }
-
-        parent::populate($values);
+        $this->channelId = $id;
+        $this->populate($this->fetchDbValues());
 
         return $this;
     }
 
-    protected function onSuccess()
+    /**
+     * Add the new channel
+     */
+    public function addChannel(): void
     {
-        if ($this->getPressedSubmitElement()->getName() === 'delete') {
-            $this->db->delete('channel', ['id = ?' => $this->channelId]);
+        $channel = $this->getValues();
 
-            return;
-        }
+        $channel['config'] = json_encode($this->filterConfig($channel['config']));
 
+        $this->db->insert('channel', $channel);
+    }
+
+    /**
+     * Edit the channel
+     *
+     * @return void
+     */
+    public function editChannel(): void
+    {
         $channel = $this->getValues();
-        $config = array_filter(
-            $channel['config'],
-            function ($configItem, $key) {
-                if (isset($this->defaultChannelOptions[$key])) {
-                    return $this->defaultChannelOptions[$key] !== $configItem;
-                }
+        $storedValues = $this->fetchDbValues();
 
-                return $configItem !== null;
-            },
-            ARRAY_FILTER_USE_BOTH
-        );
+        $channel['config'] = json_encode($this->filterConfig($channel['config']));
+        $storedValues['config'] = json_encode($this->filterConfig($storedValues['config']));
+
+        if(! empty(array_diff_assoc($channel, $storedValues))) {
+            $channel['changed_at'] = time();
```

Comment:
Both the [schema](https://github.com/Icinga/icinga-notifications/blob/24b62333d98d62042be598dc3806e323599527b4/schema/pgsql/upgrades/031.sql#L1-L3) and the [Go code](https://github.com/Icinga/icinga-notifications/blob/24b62333d98d62042be598dc3806e323599527b4/internal/channel/channel.go#L23-L24) are using millisecond precision. Writing seconds in this column results in way smaller values by a factor of 1k, resulting in a non-update from the daemon. Please use millisecond precision for all tables throughout your code.